pyccel/codegen/printing/ccode.py

```diff
@@ -1349,9 +1349,14 @@ def _print_For(self, expr):
             step  = self._print(expr.iterable.step )
         else:
             raise NotImplementedError("Only iterable currently supported is Range")
-        return ('for ({target} = {start}; {target} < {stop}; {target} += '
+        if (step[0] == "-"):
```

Comment:
A better way to test this would be `if isinstance(step, PyccelUnarySub)` that way if the printer is modified for example to add spaces the index isn't invalidated